pandar_pointcloud/src/lib/decoder/pandar_qt_decoder.cpp

```diff
@@ -87,13 +94,36 @@ void PandarQTDecoder::unpack(const pandar_msgs::PandarPacket& raw_packet)
   return;
 }
 
+PointXYZIRADT PandarQTDecoder::build_point(int block_id, int unit_id, int8_t return_type)
+{
+  const auto& block = packet_.blocks[block_id];
+  const auto& unit = block.units[unit_id];
+  double unix_second = static_cast<double>(timegm(&packet_.t));
+  PointXYZIRADT point;
+
+  double xyDistance = unit.distance * cosf(deg2rad(elev_angle_[unit_id]));
+
+  point.x = static_cast<float>(
+      xyDistance * sinf(deg2rad(azimuth_offset_[unit_id] + (static_cast<double>(block.azimuth)) / 100.0)));
+  point.y = static_cast<float>(
+      xyDistance * cosf(deg2rad(azimuth_offset_[unit_id] + (static_cast<double>(block.azimuth)) / 100.0)));
+  point.z = static_cast<float>(unit.distance * sinf(deg2rad(elev_angle_[unit_id])));
+
+  point.intensity = unit.intensity;
+  point.distance = unit.distance;
+  point.ring = unit_id;
+  point.azimuth = block.azimuth + round(azimuth_offset_[unit_id] * 100.0f);
+  point.return_type = return_type;
+  point.time_stamp = unix_second + (static_cast<double>(packet_.usec)) / 1000000.0;
+  point.time_stamp -= (static_cast<double>(block_offset_dual_[block_id] + firing_offset_[unit_id]) / 1000000.0f);
```

Comment:
The sign needs to be corrected. The block offset of QT is **positive**, but the one of 40P is **negative**. (it's very confusing !)

QT User Manual
![image](https://user-images.githubusercontent.com/8327162/113501004-4827d800-955d-11eb-9e9a-557dbffb1166.png)

40P User Manual
![image](https://user-images.githubusercontent.com/8327162/113501017-5fff5c00-955d-11eb-970b-0b09d39ca84d.png)

Reply:
Excellent, thank you and I will remedy this.